### DISCUSSION TITLE
How to install the latest version of dhcpcd in Ubuntu

### DISCUSSION THREAD

**Discussion by MadaraCx** [rsmarples/dhcpcd#212] [Q&A] [answered]
So I've been using the `apt-get install dhcpcd5` to install dhcpcd in Ubuntu(22.04.2 LTS), when I try `dhcpcd` it says the package is not available. This is the output in my machine;

```
sd@12:~$ sudo apt-get install dhcpcd
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Package dhcpcd is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source
However the following packages replace it:
  dhcpcd5

E: Package 'dhcpcd' has no installation candidate
sd@12:~$ sudo apt-get install dhcpcd5
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Suggested packages:
  dhcpcd-gtk
The following NEW packages will be installed:
  dhcpcd5
0 upgraded, 1 newly installed, 0 to remove and 229 not upgraded.
Need to get 163 kB of archives.
After this operation, 404 kB of additional disk space will be used.
Get:1 http://ie.archive.ubuntu.com/ubuntu jammy/universe amd64 dhcpcd5 amd64 7.1.0-2build1 [163 kB]
Fetched 163 kB in 0s (865 kB/s) 
Selecting previously unselected package dhcpcd5.
(Reading database ... 257926 files and directories currently installed.)
Preparing to unpack .../dhcpcd5_7.1.0-2build1_amd64.deb ...
Unpacking dhcpcd5 (7.1.0-2build1) ...
Setting up dhcpcd5 (7.1.0-2build1) ...
Processing triggers for man-db (2.10.2-1) ...

```

And fair enough when I checkout [Packages.xz](http://ie.archive.ubuntu.com/ubuntu/dists/jammy/universe/binary-amd64/) I can only see dhcpcd5 and some others like dhcpcd-gtk, dhcpcd-dbus, etc..
How do I install the latest version of dhcpcd?

**Comment by igalic** [comment, marked as answer]
you can start with https://askubuntu.com/questions/246690/how-to-use-apt-to-get-source-code-and-then-do-separate-compile

and then checkout https://packaging.ubuntu.com/singlehtml/ and http://www.debian.org/doc/manuals/maint-guide/build.en.html

or if you ignore the packaging for a second, you could just fetch the latest source here, `./configure && make && sudo make install`

**Reply by MadaraCx** [reply]
So there's no way of direct installation?

**Reply by rsmarples** [reply]
We only provide source packages, so `./configure && make && sudo make install` is the most direct approach as @igalic noted.

**Comment by rsmarples** [comment]
Here are the latest versions of dhcpcd in Ubuntu: https://packages.ubuntu.com/search?keywords=dhcpcd5

So to get newer you need to upgrade your Ubuntu. To get newer than dhcpcd-9 you'll need to install it yourself until Debian (and then Ubuntu) get around to it.

**Reply by MadaraCx** [reply]
Oh I see, thanks for the info

**Reply by igalic** [reply]
maybe someone could provide a ppa.!
